### ISSUE TITLE
Failed to build with Angular 5.0.0-beta.5

### ISSUE THREAD

**Issue by Sky4CE** [Innqube/ng2-iq-select2#50]
Due to this breaking change:
common: NgTemplateOutlet#ngOutletContext has been removed as it was deprecated since v4. Use NgTemplateOutlet#ngTemplateOutletContext instead.
![image](https://user-images.githubusercontent.com/9821633/31229811-a4fb294a-a9ea-11e7-925e-c6299fe5fede.png)

Any plans to fix this?

**Comment by diegofsza** [comment]
Sure. Will try to fix it in the following days.

**Comment by yuliankarapetkov** [comment]
Any progress on this, guys?

**Comment by egesu** [comment]
Maybe still it continues? I use Angular 5.0.0, Typescript 2.4.2, RxJS 5.5.2 (defaults of a fresh install) and I get the following error when I try to build for production (with or without the `--aot` flag, error is the same).

```
ERROR in ./node_modules/rxjs/_esm5/observable/BoundCallbackObservable.js
Module build failed: TypeError: Cannot read property 'type' of undefined
    at Object.getEffectiveTypeAnnotationNode (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:9341:17)
    at assignContextualParameterTypes (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:41652:25)
    at checkFunctionExpressionOrObjectLiteralMethod (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:41948:29)
    at checkExpressionWorker (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:42959:28)
    at checkExpression (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:42898:42)
    at checkExpressionCached (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:42779:38)
    at checkReturnStatement (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:45418:54)
    at checkSourceElement (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46763:28)
    at Object.forEach (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:1506:30)
    at checkBlock (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:44563:16)
    at checkSourceElement (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46742:28)
    at checkFunctionExpressionOrObjectLiteralMethodDeferred (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:41990:21)
    at checkDeferredNodes (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46828:25)
    at checkSourceFileWorker (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46863:17)
    at checkSourceFile (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46842:13)
    at Object.forEach (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:1506:30)
 @ ./node_modules/rxjs/_esm5/observable/bindCallback.js 2:0-68
 @ ./node_modules/rxjs/_esm5/add/observable/bindCallback.js
 @ ./node_modules/rxjs/_esm5/Rx.js
 @ ./node_modules/ng2-iq-select2/src/app/iq-select2/iq-select2.component.js
 @ ./node_modules/ng2-iq-select2/src/app/iq-select2.module.js
 @ ./src/app/app.module.ngfactory.js
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts
ERROR in ./node_modules/rxjs/_esm5/observable/BoundNodeCallbackObservable.js
Module build failed: TypeError: Cannot read property 'type' of undefined
    at Object.getEffectiveTypeAnnotationNode (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:9341:17)
    at assignContextualParameterTypes (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:41652:25)
    at checkFunctionExpressionOrObjectLiteralMethod (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:41948:29)
    at checkExpressionWorker (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:42959:28)
    at checkExpression (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:42898:42)
    at checkExpressionCached (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:42779:38)
    at checkReturnStatement (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:45418:54)
    at checkSourceElement (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46763:28)
    at Object.forEach (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:1506:30)
    at checkBlock (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:44563:16)
    at checkSourceElement (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46742:28)
    at checkFunctionExpressionOrObjectLiteralMethodDeferred (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:41990:21)
    at checkDeferredNodes (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46828:25)
    at checkSourceFileWorker (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46863:17)
    at checkSourceFile (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:46842:13)
    at Object.forEach (/home/ege/public_html/mglHq/node_modules/typescript/lib/typescript.js:1506:30)
 @ ./node_modules/rxjs/_esm5/observable/bindNodeCallback.js 2:0-76
 @ ./node_modules/rxjs/_esm5/add/observable/bindNodeCallback.js
 @ ./node_modules/rxjs/_esm5/Rx.js
 @ ./node_modules/ng2-iq-select2/src/app/iq-select2/iq-select2.component.js
 @ ./node_modules/ng2-iq-select2/src/app/iq-select2.module.js
 @ ./src/app/app.module.ngfactory.js
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts
```

**Comment by gplume** [comment]
Hi, I can report the exact same errors as @egesu with the only exception that it builds and works correctly -without- `--aot`
typescript `2.6.1`

**Comment by egesu** [comment]
It's fixed with PR #53, but there is no version number for this.

@saithis @diegofsza can you publish a new version to npm?

**Comment by diegofsza** [comment]
Hi @egesu, @gplume , I just released version 7.0.0 including PR #53.

**Comment by gplume** [comment]
@diegofsza Thank you very much!
Works perfect now with @angular-5.0.1
Great Module btw !
